packages/core/src/Site/index.js

```diff
@@ -706,80 +719,90 @@ class Site {
+    return await fs.remove(this.tempPath);
   }
 
-  _rebuildSourceFiles() {
+  async _rebuildSourceFiles() {
     logger.info('Page added or removed, updating list of site\'s pages...');
     this.beforeSiteGenerate();
 
     this.layoutManager.removeLayouts();
 
     const removedPageFilePaths = this.updateAddressablePages();
-    return this.removeAsset(removedPageFilePaths)
-      .then(() => {
-        if (this.onePagePath) {
-          this.mapAddressablePagesToPages(this.addressablePages || [], this.getFavIconUrl());
+    try {
+      await this.removeAsset(removedPageFilePaths)
+      return await (async () => {
+          if (this.onePagePath) {
+            this.mapAddressablePagesToPages(this.addressablePages || [], this.getFavIconUrl());
 
-          return this.rebuildPageBeingViewed(this.currentPageViewed)
-            .then(() => this.lazyBuildAllPagesNotViewed());
-        }
+            await this.rebuildPageBeingViewed(this.currentPageViewed);
+            return await this.lazyBuildAllPagesNotViewed();
+          }
 
-        logger.warn('Rebuilding all pages...');
-        return this.buildSourceFiles();
-      })
-      .catch(error => Site.rejectHandler(error, [this.tempPath, this.outputPath]));
+          logger.warn('Rebuilding all pages...');
+          return this.buildSourceFiles();
+      })();
+    } catch(error) {
+      return Site.rejectHandler(error, [this.tempPath, this.outputPath]);
+    }
   }
 
-  _buildMultipleAssets(filePaths) {
+  async _buildMultipleAssets(filePaths) {
     const filePathArray = Array.isArray(filePaths) ? filePaths : [filePaths];
     const uniquePaths = _.uniq(filePathArray);
     const fileIgnore = ignore().add(this.siteConfig.ignore);
     const fileRelativePaths = uniquePaths.map(filePath => path.relative(this.rootPath, filePath));
     const copyAssets = fileIgnore.filter(fileRelativePaths)
       .map(asset => fs.copy(path.join(this.rootPath, asset), path.join(this.outputPath, asset)));
-    return Promise.all(copyAssets)
-      .then(() => logger.info('Assets built'));
+    await Promise.all(copyAssets);
+    return logger.info('Assets built');
   }
 
-  _removeMultipleAssets(filePaths) {
+  async _removeMultipleAssets(filePaths) {
     const filePathArray = Array.isArray(filePaths) ? filePaths : [filePaths];
     const uniquePaths = _.uniq(filePathArray);
     const fileRelativePaths = uniquePaths.map(filePath => path.relative(this.rootPath, filePath));
     const filesToRemove = fileRelativePaths.map(
       fileRelativePath => path.join(this.outputPath, fileRelativePath));
     const removeFiles = filesToRemove.map(asset => fs.remove(asset));
-    return removeFiles.length === 0
-      ? Promise.resolve('')
-      : Promise.all(removeFiles)
-        .then(() => logger.debug('Assets removed'));
+    if (removeFiles.length === 0) {
+      Promise.resolve('');
+    } else {
+      await Promise.all(removeFiles);
+      return logger.debug('Assets removed');
```

Comment:
no `return` needed here, could remove the top branch as well since its now async

Reply:
Removed.